Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Policies/Edge_collapse/GarlandHeckbert_probabilistic_policies.h

```diff
@@ -0,0 +1,149 @@
+#ifndef CGAL_SURFACE_MESH_SIMPLIFICATION_POLICIES_GARLANDHECKBERT_PROBABILISTIC_POLICIES_H
+#define CGAL_SURFACE_MESH_SIMPLIFICATION_POLICIES_GARLANDHECKBERT_PROBABILISTIC_POLICIES_H
+
+#include <CGAL/license/Surface_mesh_simplification.h>
+
+#include <CGAL/Surface_mesh_simplification/internal/Common.h>
+#include <CGAL/Surface_mesh_simplification/Policies/Edge_collapse/internal/GarlandHeckbert_core.h>
+#include <CGAL/Surface_mesh_simplification/Policies/Edge_collapse/internal/GarlandHeckbert_policy_base.h>
+
+#include <Eigen/Dense>
+
+namespace CGAL {
+namespace Surface_mesh_simplification {
+
+// forward-declare template class
+template<typename TriangleMesh, typename GeomTraits>
+class GarlandHeckbert_policies;
+
+template<typename TriangleMesh, typename GeomTraits>
+using Cost_property = CGAL::dynamic_vertex_property_t<Eigen::Matrix<typename GeomTraits::FT, 4, 4,
```

Comment:
Avoid typedefing in namespaces; especially with such generic typedefs.

Reply:
How about plugging the typedefs into an unnamed namespace? That should keep the code (somewhat) readable but then the types are not be visible from outside.